pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -151,15 +152,32 @@ func (r *CloudEntitySelectorReconciler) processCreateOrUpdate(selector *cloudv1a
 	}
 
 	r.selectorToAccountMap[*selectorNamespacedName] = *accountNamespacedName
+
+	callCESDelete := false
+	defer func() {
+		if callCESDelete {
+			_ = r.processDelete(selectorNamespacedName)
+		}
+	}()
+
 	err = cloudInterface.AddAccountResourceSelector(accountNamespacedName, selector)
 	if err != nil {
-		_ = r.processDelete(selectorNamespacedName)
+		r.Log.Info(errorMsgSelectorAddFail, "selector", selectorNamespacedName, "error", err)
+		callCESDelete = true
```

Comment:
can you simulate errors for ces delete failures and verify readd of ces is working fine

Reply:
Yes , I had to simulate an error through a code hack and processDelete for CES gets called on reconciler error. As this is a reconciler error, reconciler will be called again for CES add. This time it goes through and vm inventory is pulled base don CES selector.

2023-03-01T16:39:59.823Z        INFO        controllers.CloudEntitySelector        cloud/cloudentityselector_controller.go:75        Test: simulate fail
2023-03-01T16:39:59.823Z        INFO        controllers.CloudEntitySelector        cloud/cloudentityselector_controller.go:161        Received request        {"selector": "sample-ns/cloudentityselector-azure-sample", "operation": "delete"}
2023-03-01T16:39:59.823Z        INFO        poller.AccountPoller        cloud/cloudentityselector_controller.go:218        Restarting account poller        {"account": "sample-ns/cloudprovideraccount-azure-sample"}
2023-03-01T16:39:59.823Z        ERROR        setup        controller/controller.go:273        Reconciler error        {"controller": "cloudentityselector", "controllerGroup": "crd.cloud.antrea.io", "controllerKind": "CloudEntitySelector", "cloudEntitySelector": {"name":"cloudentityselector-azure-sample","namespace":"sample-ns"}, "namespace": "sample-ns", "name": "cloudentityselector-azure-sample", "reconcileID": "65f13121-6fc0-4716-adb4-e0a1f5cb31ea", "error": "selector add failed, poller is not created for account cloudentityselector-azure-sample, err: <nil>"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
/tmp/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
/tmp/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:234
2023-03-01T16:39:59.829Z        INFO        controllers.CloudEntitySelector        cloud/cloudentityselector_controller.go:75        Received request        {"selector": "sample-ns/cloudentityselector-azure-sample", "operation": "create/update"}
2023-03-01T16:39:59.846Z        INFO        controllers.VirtualMachine        manager/runnable_group.go:219        Init done        {"controller": "VirtualMachine"}
2023-03-01T16:40:00.900Z        DEBUG        azure-plugin        azure/azure_compute.go:228        fetching vm resources from cloud skipped        {"account": "sample-ns/cloudprovideraccount-azure-sample", "resource-filters": "not-configured"}
2023-03-01T16:40:00.901Z        DEBUG        azure-plugin        internal/services_common.go:124        cached vpcs        {"service": "COMPUTE", "account": "sample-ns/cloudprovideraccount-azure-sample", "vpc objects": 35}
2023-03-01T16:40:00.901Z        INFO        azure-plugin        azure/azure_query_helpers.go:33        selector stats        {"VpcIdOnlyMatch": 1, "VpcIdWithOtherMatches": 0, "VmIdOnlyMatches": 0, "VmIdAndVmNameMatches": 0, "VmNameOnlyMatches": 0}
2023-03-01T16:40:00.901Z        INFO        poller.AccountPoller        wait/wait.go:155        Discovered compute resources statistics        {"account": "sample-ns/cloudprovideraccount-azure-sample", "vpcs": 35, "virtual-machines": 0}
2023-03-01T16:40:00.901Z        INFO        poller.AccountPoller        cloud/cloudentityselector_controller.go:186        Restarting account poller        {"account": "sample-ns/cloudprovideraccount-azure-sample"}
2023-03-01T16:40:00.901Z        INFO        controllers-sync-status        cloud/cloudentityselector_controller.go:131        Sync done        {"controller": "CloudEntitySelector"}
2023-03-01T16:40:01.962Z        DEBUG        azure-plugin        azure/azure_compute.go:228        fetching vm resources from cloud        {"account": "sample-ns/cloudprovideraccount-azure-sample", "resource-filters": "configured"}
2023-03-01T16:40:02.311Z        DEBUG        azure-plugin        azure/azure_compute.go:228        vm instances from cloud        {"service": "COMPUTE", "account": "sample-ns/cloudprovideraccount-azure-sample", "instances": 1}
2023-03-01T16:40:02.312Z        DEBUG        azure-plugin        internal/services_common.go:124        cached vpcs        {"service": "COMPUTE", "account": "sample-ns/cloudprovideraccount-azure-sample", "vpc objects": 35}
2023-03-01T16:40:02.312Z        DEBUG        azure-plugin        azure/azure_compute.go:266        cached vm instances        {"service": "COMPUTE", "account": "sample-ns/cloudprovideraccount-azure-sample", "instances": 1}
2023-03-01T16:40:02.312Z        DEBUG        azure-plugin        internal/services_common.go:106        CRDs        {"service": "COMPUTE", "account": "sample-ns/cloudprovideraccount-azure-sample", "virtual-machine CRDs": 1}